packages/tailwindcss/src/utilities.ts

```diff
@@ -29,38 +28,35 @@ type SuggestionDefinition =
     }
 
 export class Utilities {
-  private utilities = new Map<
-    string | symbol,
+  private arbitraryFn!: CompileFn<'arbitrary'>
+  private utilities = new DefaultMap<
+    string,
     {
-      kind: Candidate['kind']
+      kind: 'static' | 'functional'
       compileFn: CompileFn<any>
-    }
-  >()
+    }[]
+  >(() => [])
 
   private completions = new Map<string, () => SuggestionGroup[]>()
 
   static(name: string, compileFn: CompileFn<'static'>) {
-    this.set(name, { kind: 'static', compileFn: compileFn })
+    this.utilities.get(name).push({ kind: 'static', compileFn })
   }
 
   functional(name: string, compileFn: CompileFn<'functional'>) {
-    this.set(name, { kind: 'functional', compileFn: compileFn })
+    this.utilities.get(name).push({ kind: 'functional', compileFn })
   }
 
   arbitrary(compileFn: CompileFn<'arbitrary'>) {
-    this.set(ARBITRARY_VARIANT, { kind: 'arbitrary', compileFn: compileFn })
-  }
-
-  has(name: string) {
-    return this.utilities.has(name)
+    this.arbitraryFn = compileFn
   }
 
-  get(name: string | symbol) {
-    return this.utilities.get(name)
+  has(name: string, kind: 'static' | 'functional') {
+    return this.utilities.has(name) && this.utilities.get(name).some((fn) => fn.kind === kind)
```

Comment:
Q: Does it make sense to store the utilities on a per-kind basis? This way we can do the lookup once and don't need to do the `.some()` part ever 🤔

Reply:
We did that first but:
- It looked awful
- You lose potential ordering between multiple definitions of utilities